encoding/jsonbq/marshal.go

```diff
@@ -0,0 +1,47 @@
+package jsonbq
+
+import (
+	"cloud.google.com/go/bigquery"
+	"encoding/json"
+)
+
+const (
+	ModeRepeated string = "REPEATED"
+	ModeRequired string = "REQUIRED"
+	ModeNullable string = "NULLABLE"
+)
+
+// Field describes the schema of a field in BigQuery.
+type Field struct {
+	Name        string   `json:"name"`
+	Type        string   `json:"type"`
+	Mode        string   `json:"mode"`
+	Description string   `json:"description,omitempty"`
+	Fields      []*Field `json:"fields,omitempty"`
+}
+
+func MarshalSchema(schema bigquery.Schema) ([]byte, error) {
+	fields := convertSchemaToMarshableFields(schema)
+	return json.Marshal(fields)
+}
+
+func convertSchemaToMarshableFields(schema bigquery.Schema) []*Field {
+	fields := make([]*Field, len(schema))
+	for i, fieldSchema := range schema {
```

Comment:
extract this into a separate function `convertField`?

Reply:
Yes!